httphandlers/analyses.go

```diff
@@ -0,0 +1,72 @@
+package httphandlers
+
+import (
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+)
+
+type AnalysisStatus struct {
+	Status string `json:"status"`
+}
+
+// AnalysisStatusByExternalID returns the status of an analysis identified by its external ID.
+//
+//	@ID				analysis-status-by-external-id
+//	@Summary		Returns the status of an analysis
+//	@Description	Returns the status of an analysis identified by its external ID.
+//	@Param			external-id	path		string	true	"external ID"
+//	@Success		200			{object}	AnalysisStatus
+//	@Failure		400			{object}	common.ErrorResponse
+//	@Failure		500			{object}	common.ErrorResponse
+//	@Router			/info/analysis/status/by/external-id/{external-id} [get]
+func (h *HTTPHandlers) AnalysisStatusByExternalID(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	externalID := c.Param("external-id")
+	if externalID == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "external-id parameter is empty")
+	}
+
+	analysisID, err := h.apps.GetAnalysisIDByExternalID(ctx, externalID)
```

Comment:
It might be helpful to have a bit of code that looks for `ErrNoRows` and returns a 404 in that case, just to make it clear that the analysis just wasn't found.

Reply:
I pushed this change.